### ISSUE TITLE
wwclient fails to start due to existing process

### ISSUE THREAD

**Issue by franciscoadasme** [warewulf/warewulf#1066]
### Steps to reproduce

Boot up a compute note and check the wwclient service's status

### Error message

The `wwclient` service fails to start reporting that it found an existing PIDFile, but it doesn't exist.

```console
$ journalctl -u wwclient
Jan 31 14:53:19 rose systemd[1]: Starting Warewulf node runtime overlay update...
Jan 31 14:53:19 rose wwclient[2837]: Error: found pidfile /var/run/wwclient.pid not starting
Jan 31 14:53:19 rose wwclient[2837]: ERROR: found pidfile /var/run/wwclient.pid not starting
Jan 31 14:54:19 rose systemd[1]: wwclient.service: start operation timed out. Terminating.
Jan 31 14:54:19 rose systemd[1]: wwclient.service: Failed with result 'timeout'.
Jan 31 14:54:19 rose systemd[1]: Failed to start Warewulf node runtime overlay update.
$ ps cax | grep ww
$ ls -lh /var/run/wwclient.pid
ls: cannot access '/var/run/wwclient.pid': No such file or directory
```

If I start it manually, there is no issue:

```console
$ systemctl start wwclient
$ systemctl status wwclient
● wwclient.service - Warewulf node runtime overlay update
     Loaded: loaded (/etc/systemd/system/wwclient.service; enabled; preset: disabled)
     Active: active (running) since Wed 2024-01-31 15:09:43 UTC; 8s ago
   Main PID: 3531 (wwclient)
      Tasks: 9 (limit: 3292209)
     Memory: 5.8M
        CPU: 11ms
     CGroup: /system.slice/wwclient.service
             └─3531 /warewulf/wwclient

Jan 31 15:09:38 rose systemd[1]: Starting Warewulf node runtime overlay update...
Jan 31 15:09:38 rose wwclient[3531]: 2024/01/31 15:09:38 Updating live file system LIVE, cancel now if this is in error
Jan 31 15:09:43 rose wwclient[3531]: 2024/01/31 15:09:43 Updating system
Jan 31 15:09:43 rose systemd[1]: Started Warewulf node runtime overlay update.
```

or even

```console
$ /warewulf/wwclient
2024/01/31 15:05:53 Updating live file system LIVE, cancel now if this is in error
2024/01/31 15:05:58 Updating system
2024/01/31 15:06:58 Updating system
```

I'm relatively new to this stuff so I don't know where else to look for the issue.

### Information on your system

I'm using Warewulf v4.5 nightly build (commit adad227) with Rocky Linux 9.3, and I'm provisioning a single node with default settings and overlays.

```console
$ wwctl version
wwctl version:   4.5.x-1
rpc version: apiPrefix:"rc1"  apiVersion:"1"  warewulfVersion:"4.5.x-1"
$ cat /etc/os-release
NAME="Rocky Linux"
VERSION="9.3 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.3"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.3 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.3"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.3"
```

### General information

- [X] I have run `wwctl version` and reported the contents of `/etc/os-release`
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have captured and reported relevant error messages and logs

**Comment by anderbubble** [comment]
Thanks for the report, @franciscoadasme. I admit, I have been testing mostly on the server side recently; but I'll install the nightly on my test cluster and see if I can replicate your experience.

**Comment by anderbubble** [comment]
Confirmed.

```
[root@wwnode1 ~]# journalctl -u wwclient.service
Feb 07 06:43:45 wwnode1 systemd[1]: Starting Warewulf node runtime overlay update...
Feb 07 06:43:45 wwnode1 wwclient[345]: Error: found pidfile /var/run/wwclient.pid not starting
Feb 07 06:43:45 wwnode1 wwclient[345]: ERROR: found pidfile /var/run/wwclient.pid not starting
[root@wwnode1 ~]# systemctl status wwclient.service
● wwclient.service - Warewulf node runtime overlay update
     Loaded: loaded (/etc/systemd/system/wwclient.service; enabled; preset: disabled)
     Active: activating (start) since Wed 2024-02-07 06:43:45 UTC; 35s ago
   Main PID: 187 (wwclient)
      Tasks: 0 (limit: 60211)
     Memory: 4.0K
        CPU: 14ms
     CGroup: /system.slice/wwclient.service
             ‣ 187 /warewulf/wwclient

Feb 07 06:43:45 wwnode1 systemd[1]: Starting Warewulf node runtime overlay update...
Feb 07 06:43:45 wwnode1 wwclient[345]: Error: found pidfile /var/run/wwclient.pid not starting
Feb 07 06:43:45 wwnode1 wwclient[345]: ERROR: found pidfile /var/run/wwclient.pid not starting
[root@wwnode1 ~]# systemctl status wwclient.service
● wwclient.service - Warewulf node runtime overlay update
     Loaded: loaded (/etc/systemd/system/wwclient.service; enabled; preset: disabled)
     Active: activating (start) since Wed 2024-02-07 06:43:45 UTC; 43s ago
   Main PID: 187 (wwclient)
      Tasks: 0 (limit: 60211)
     Memory: 4.0K
        CPU: 14ms
     CGroup: /system.slice/wwclient.service
             ‣ 187 /warewulf/wwclient

Feb 07 06:43:45 wwnode1 systemd[1]: Starting Warewulf node runtime overlay update...
Feb 07 06:43:45 wwnode1 wwclient[345]: Error: found pidfile /var/run/wwclient.pid not starting
Feb 07 06:43:45 wwnode1 wwclient[345]: ERROR: found pidfile /var/run/wwclient.pid not starting
[root@wwnode1 ~]# pgrep -lf wwclient
[root@wwnode1 ~]# ls /var/run/wwclient.pid
ls: cannot access '/var/run/wwclient.pid': No such file or directory
[root@wwnode1 ~]# systemctl status wwclient.service
× wwclient.service - Warewulf node runtime overlay update
     Loaded: loaded (/etc/systemd/system/wwclient.service; enabled; preset: disabled)
     Active: failed (Result: timeout) since Wed 2024-02-07 06:44:45 UTC; 17s ago
    Process: 187 ExecStart=/warewulf/wwclient (code=exited, status=0/SUCCESS)
   Main PID: 187 (code=exited, status=0/SUCCESS)
        CPU: 14ms

Feb 07 06:43:45 wwnode1 systemd[1]: Starting Warewulf node runtime overlay update...
Feb 07 06:43:45 wwnode1 wwclient[345]: Error: found pidfile /var/run/wwclient.pid not starting
Feb 07 06:43:45 wwnode1 wwclient[345]: ERROR: found pidfile /var/run/wwclient.pid not starting
Feb 07 06:44:45 wwnode1 systemd[1]: wwclient.service: start operation timed out. Terminating.
Feb 07 06:44:45 wwnode1 systemd[1]: wwclient.service: Failed with result 'timeout'.
Feb 07 06:44:45 wwnode1 systemd[1]: Failed to start Warewulf node runtime overlay update.
```

**Comment by JasonYangShadow** [comment]
yeah confirmed as well with the latest main branch nightly build
![image](https://github.com/warewulf/warewulf/assets/2051711/d53117f1-39e9-4f34-9b99-14eb24bfbb7a)

**Comment by franciscoadasme** [comment]
I confirm that this issue is fixed as of release [v4.5.0rc2](https://github.com/warewulf/warewulf/releases/tag/v4.5.0rc2) (commit 1c9412b).

Thank you.

**Comment by anderbubble** [comment]
Thanks, @franciscoadasme!
